src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java

```diff
@@ -212,38 +214,49 @@ else if (value instanceof Array) {
         // finish the file
         writer.close();
 
+        String replacedSql = markupSql(instanceId, whereClause, orderClause, definition.getFilenameTemplate().getSql());
         // get filename
-        Object filename = evaluateSql(instanceId, whereClause, orderClause, definition.getFilenameTemplate().getSql());
+        Object filename = evaluateSql(replacedSql);
 
         // this is awful...
         // format the file into the output stream
-        InputStream is = createWritable.getStream();
-        outputFormatter.write((String)filename, is);
-        is.close();
+        if (filename != null) {
+            InputStream is = createWritable.getStream();
+            outputFormatter.write((String) filename, is);
+            is.close();
```

Comment:
I know you didn't write this - but we I just noticed it and we should get into the habit of always ensuring input streams are closed by wrapping in a try resources block .  e.g.


    try (InputStream is = createWritable.getStream()) {
        outputFormatter.write((String) filename, is);
    }